stix2/datastore/filesystem.py

```diff
@@ -3,13 +3,362 @@
 
 """
 
+import errno
 import json
 import os
+import stat
 
+import pytz
+import six
+
+from stix2.base import _STIXBase
 from stix2.core import Bundle, parse
 from stix2.datastore import DataSink, DataSource, DataStoreMixin
 from stix2.datastore.filters import Filter, FilterSet, apply_common_filters
-from stix2.utils import deduplicate, get_class_hierarchy_names
+from stix2.utils import get_type_from_id, is_marking
+
+
+def _timestamp2filename(timestamp):
```

Comment:
Are **any** these methods generic enough or have the ability to be used for other datastores? If this is the case, could we move them to `stix2.utils`, `stix2.datastores.__init__`  or a better location.

Reply:
I did decouple things so that the design felt more like a set of integrated-but-independently-usable and testable pieces (and I did independently test them in the unit tests).  But I think most wouldn't have much applicability outside the filesystem store.  No need for a general white/blacklist class, is there?  The ability to scan a directory in the context of a white/blacklist?  Did you have a particular class/function in mind, that you thought would be useful?

Reply:
Definitely no need (that I know of) for a generic while/blacklist class or any functionality. Just wondering if there was anything that was re-usable from this code then it would make sense to move it into a move convenient place..